### DISCUSSION TITLE
python-can library has problems reading messages with arbitration id (10 digits), and of type extended can

### DISCUSSION THREAD

**Discussion by masoomk87** [hardbyte/python-can#1493] [Q&A] [answered]
Hi

I have been trying to use python-can library to read Vector logged binary logging files (.blf) saved on my disk. I face a strange issue

1. Message with arbitration id less than 10 digits (I have few of them), and having the property is_extended_id=False, are read out properly
2. Message with arbitration id of 10 digits, and having the property is_extended_id=True, cannot be read. It actually generates strange arbitration id after parsing the .blf files which could not be traced back in the .dbc file associated with the log

Does anyone knows what could be going wrong ?

I am using the CANALYZER to log the multiple CAN channels (6 to be precise) and make a .blf file out of it

**Comment by zariiii9003** [comment]
Can you share the file?

**Reply by j-c-cook** [reply]
ping @masoomk87

**Comment by masoomk87** [comment, marked as answer]
I found that in the CAN blf reader module - https://python-can.readthedocs.io/en/master/_modules/can/io/blf.html, the can arbitration ID is define below

![image](https://user-images.githubusercontent.com/58339669/215848539-6a5d607a-6d4b-4499-a682-c9bd257e9630.png)

Implementing the same in the code solved the issue. I am not sure why the 0x1FFFFFFF, are AND with CAN arbitration ID

**Reply by grant-allan-ctct** [reply]
On the question of "why the `0x1FFFFFFF`?": I am going to suggest that it's because the arbitration ID is 29 bits for an extended CAN frame.  By performing a bitwise AND of the given `can_id` value with `0x1FFFFFFF`, it ensures that it's not going to "overflow" the allowable 29 bits.  

In `0x1FFFFFFF`, every "F" represents four bits, and there are seven of those (making 28 bits so far), but then there is one additional bit upstream of that - total of 29.
